powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/converter/v3/RecoveryPukConverter.java

```diff
@@ -81,6 +81,7 @@ public String fromDBValue(RecoveryPuk pukHash, long applicationId, String userId
         final EncryptionMode encryptionMode = pukHash.getEncryptionMode();
         if (encryptionMode == null) {
             logger.error("Missing key encryption mode");
+            // Rollback is not required, error occurs before writing to database
```

Comment:
Good catch, yes, let's describe the behavior im Javadoc.

Reply:
By the way, the point of copy-pasting the comments to all exception methods is for future developers, who may not have experienced the current bug and the comments will make them thing about behavior of the code from database rollback perspective. If you think it is a bit overwhelming, we can remove the comments, but it was intentional because of the potential reintroduction of the bug.